pkg/controller/controller_test.go

```diff
@@ -735,6 +797,30 @@ func TestHandleUpdateClusterUpdatesStatefulSets(t *testing.T) {
 			increaseReplicas:      true,
 			expUpdateStatefulSets: []string{},
 		},
+		{
+			name: "updates stateful sets does not progress if pods not updated",
+			cluster: newMeta("cluster1", map[string]string{
+				"foo":                      "bar",
+				"operator.m3db.io/app":     "m3db",
+				"operator.m3db.io/cluster": "cluster1",
+			}, nil),
+			sets: []*metav1.ObjectMeta{
+				newMeta("cluster1-rep0", nil, map[string]string{
+					annotations.Update: "enabled",
+				}),
+				newMeta("cluster1-rep1", nil, map[string]string{
+					annotations.Update: "enabled",
+				}),
+				newMeta("cluster1-rep2", nil, map[string]string{
+					annotations.Update: "enabled",
+				}),
+			},
+			newImage:                    "m3db:v2.0.0",
+			simulatePodsNotUpdated:      true,
+			expUpdateStatefulSets:       []string{"cluster1-rep0", "cluster1-rep1", "cluster1-rep2"},
+			expFailedUpdateStatefulSets: []string{"cluster1-rep1", "cluster1-rep2"},
+			expNotDone:                  true,
```

Comment:
for my own understanding: this would be false before your change?

Reply:
@wesleyk it was more that, before this change if `simulatePodsNotUpdated` had existed in this test code and the "UpdatedReplicas" did not match the expected replicas in the statefulset it would have progressed to update the next statefulset anyway.